src/Platform/Microsoft.Testing.Platform/Hosts/CommonTestHost.cs

```diff
@@ -69,6 +75,9 @@ public async Task<int> RunAsync()
         }
         finally
         {
+            // Dispose the activity
+            activity?.Dispose();
```

Comment:
I think it should be the last one...we could in the dispose have some activities "to dispose"